src/features/tokenGatedRooms/helpers/createTokenGatePolicy.ts

```diff
@@ -0,0 +1,197 @@
+        address: Address
+    } | null
+> = {
+    [TokenStandard.ERC1155]: {
+        abi: erc1155abi,
+        address: ERC1155PolicyFactoryAddress,
+    },
+    [TokenStandard.ERC20]: {
+        abi: erc20abi,
+        address: ERC20PolicyFactoryAddress,
+    },
+    [TokenStandard.ERC721]: {
+        abi: erc721abi,
+        address: ERC721PolicyFactoryAddress,
+    },
+    [TokenStandard.ERC777]: {
+        abi: erc777abi,
+        address: ERC777PolicyFactoryAddress,
+    },
+    [TokenStandard.Unknown]: null,
+}
+
+interface Params {
+    owner: Address
+    tokenAddress: Address
+    tokenType: TokenType
+    roomId: RoomId
+    minRequiredBalance?: string
+    tokenIds: string[]
+    stakingEnabled: boolean
+    provider: Provider
+    streamrClient: StreamrClient
+}
+
+export default function createTokenGatePolicy({
+    owner: requester,
+    tokenAddress,
+    tokenType,
+    roomId,
+    minRequiredBalance,
+    tokenIds,
+    stakingEnabled,
+    provider,
+    streamrClient,
+}: Params) {
+    return spawn(function* () {
+        yield race([
+            take(WalletAction.changeAccount),
+            call(function* () {
+                let tc: Controller | undefined
+
+                let dismissToast = false
+
+                try {
+                    tc = yield retoast(tc, {
+                        title: 'Deploying token gate…',
+                        type: ToastType.Processing,
+                    })
+
+                    dismissToast = true
+
+                    yield preflight({
+                        provider,
+                        requester,
+                    })
+
+                    const factory = Factory[tokenType.standard]
+
+                    if (!factory) {
+                        throw new Error(`Unsupported standard: ${tokenType.standard}`)
+                    }
+
+                    const { abi, address } = factory
+
+                    const factoryContract = new Contract(
+                        address,
+                        abi,
+                        new providers.Web3Provider(provider).getSigner()
+                    )
+
+                    const tx: Record<string, any> = yield factoryContract.create(
+                        tokenAddress,
+                        roomId,
+                        minRequiredBalance,
+                        tokenIds,
+                        stakingEnabled,
+                        [PermissionType.Publish, PermissionType.Subscribe]
+                    )
+
+                    yield tx.wait(10)
+
+                    const policyRegistry = getPolicyRegistry(provider)
+
+                    let policyAddress: Address = ZeroAddress
+
+                    tc = yield retoast(tc, {
+                        title: 'Waiting for the network…',
+                        type: ToastType.Processing,
+                    })
+
+                    dismissToast = true
+
+                    yield retry(30, 1000, async function () {
```

Comment:
30 tries with 1s between each failure - replaces the `waitForPolicyToBeDeployed`.